erts/emulator/beam/jit/beam_jit_common.cpp

```diff
@@ -313,6 +253,120 @@ void BeamModuleAssembler::codegen(char *buff, size_t len) {
                            CodeHolder::kCopyPadSectionBuffer);
 }
 
+void BeamModuleAssembler::register_metadata(const BeamCodeHeader *header) {
+#ifndef WIN32
+    const BeamCodeLineTab *line_table = header->line_table;
+
+    char name_buffer[MAX_ATOM_SZ_LIMIT];
+    std::string module_name = getAtom(mod);
+    std::vector<AsmRange> ranges;
```

Comment:
Should we add a check here to return if no metadata is to be updated? So that we don't build the AsmRange structure in vain. My original thinking was that there was no need as when profiling this function did not show up, but maybe it is time to do something about it?

Reply:
Ranges and lines improve core dumps, so they're more or less always needed. I'm not sure it's worth adding an option to control it.